app/controllers/api_controller/reports.rb

```diff
@@ -44,5 +50,50 @@ def import_resource_reports(_type, _id, data)
       result, meta = MiqReport.import_from_hash(data["report"], options)
       action_result(meta[:level] == :info, meta[:message], :result => result)
     end
+
+    def schedule_resource_reports(type, id, data)
+      api_action(type, id) do |klass|
+        report = resource_search(id, type, klass)
+        schedule_reports(report, type, id, data)
+      end
+    end
+
+    private
+
+    def schedule_reports(report, type, id, data)
+      desc = "scheduling of report #{report.id}"
+      schedule = report.add_schedule fetch_schedule_data(data)
+      res = action_result(true, desc)
+      add_report_schedule_to_result(res, schedule.id, report.id)
+      add_href_to_result(res, type, id)
+      res
+    rescue => err
+      action_result(false, err.to_s)
+    end
+
+    def fetch_schedule_data(data)
+      schedule_data = data.except(*SCHEDULE_BODY_ATTR)
```

Comment:
OK, I follow now. I think a verboser constant name, perhaps `SCHEDULE_ATTRS_TO_TRANSFORM` or similar.

OTOH, did you consider designing the contract such that they arrive in the structure that you need?